bundles/admin/admin-announcements/publisher/AnnouncementsTool.js

```diff
@@ -0,0 +1,357 @@
+        },
+
+        /**
+        * Get tool object.
+        * @method getTool
+        *
+        * @returns {Object} tool description
+        */
+        getTool: function () {
+            return {
+                id: 'Oskari.framework.announcements.plugin.AnnouncementsPlugin',
+                title: 'AnnouncementsPlugin',
+                config: {}
+            };
+        },
+
+        /**
+        * Is the tool toggled on by default.
+        * @method isDefaultTool
+        * @public
+        *
+        * @returns {Boolean} is the tool toggled on by default.
+        */
+        isDefaultTool: function () {
+            return false;
+        },
+
+        /**
+        * Get extra options.
+        * @method getExtraOptions
+        * @public
+        *
+        * @returns {Object} jQuery element
+        */
+        getExtraOptions: function () {
+            const me = this;
+            const buttonLabel = me.localization.tool.buttonLabel;
+            const template = me.templates.announcements.clone();
+
+            // Set the button handler
+            template.find('button').html(buttonLabel).on('click', function () {
+                if (me.isAnnouncementsDialogOpen === false) {
+                    me._openAnnouncementsDialog(jQuery(this));
+                }
+            });
+            if (me.annTitles.length > 0) {
+                template.find('input[name=publisher-announcements]').val(me.annTitles.toString());
+            }
+            return template;
+        },
+
+        /**
+         * Creates and opens the dialog from which to choose the announcements.
+         *
+         * @method _openAnnouncementsDialog
+         */
+        _openAnnouncementsDialog: function () {
+            const me = this;
+            const popup = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+            const closeButton = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            const title = me.localization.tool.popup.title;
+            const content = me.templates.announcementsPopup.clone();
+
+            closeButton.setTitle(me.localization.tool.popup.close);
+            closeButton.setHandler(function () {
+                popup.close(true);
+                me.isAnnouncementsDialogOpen = false;
+            });
+
+            me.announcements.forEach(announcement => {
+                const announcementInput = me.templates.inputCheckbox.clone();
+                const annName = announcement.title;
+                const annTime = announcement.begin_date.replace(/-/g, '/') + ' - ' + announcement.end_date.replace(/-/g, '/');
+
+                announcementInput.find('input[type=checkbox]').attr({
+                    'id': announcement.id,
+                    'name': 'announcement',
+                    'value': announcement.title
+                });
+                announcementInput.find('label').html(annName).attr({
+                    'for': announcement.id
+                });
```

Comment:
If you must use id's, prefix them with something to make it less likely they will conflict with id's on the rest of the page: 

```suggestion
const idPrefix = 'oskari_announcement_select_';
                announcementInput.find('input[type=checkbox]').attr({
                    'id': idPrefix  + announcement.id,
                    'name': 'announcement',
                    'value': announcement.title
                });
                announcementInput.find('label').html(annName).attr({
                    'for': idPrefix  + announcement.id
                });
```

Also since "name" is always "announcement" you should move it to the template.